ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -150,17 +149,7 @@ public void onOpen(Session session, EndpointConfig endpointConfig) {
         return;
       }
 
-      queryMetadata.getKafkaStreams().setUncaughtExceptionHandler((thread, e) -> {
-        log.error("streams exception in session {}", session.getId(), e);
-        closeSession(session, new CloseReason(
-            CloseCodes.UNEXPECTED_CONDITION,
-            "streams exception"
-        ));
-      });
-      log.info("Running query {}", queryMetadata.getQueryApplicationId());
-      queryMetadata.getKafkaStreams().start();
-
-      future = exec.scheduleWithFixedDelay(() -> {
+      final Runnable drainQueue = () -> {
```

Comment:
This can run from multiple threads at the same time now. Do we need to synchronize anything? What happens if two threads call sendText?

Reply:
agree reordering is possible, let me change it to enforce ordering, and that should solve everything.

Reply:
@rodesai ok should be fixed now